marshal/proto_test.go

```diff
@@ -0,0 +1,142 @@
+	}
+	assert.Equal(t, 3, len(schemaHandler.ValueColumns))
+	assert.Equal(t, 6, len((*tableMap)["Parquet_go_root\x01Timestamp"].Values))
+	assert.Equal(t, int64(1001), (*tableMap)["Parquet_go_root\x01Timestamp"].Values[0].(int64))
+	assert.Equal(t, int64(2001), (*tableMap)["Parquet_go_root\x01Timestamp"].Values[1].(int64))
+	assert.Equal(t, int64(3001), (*tableMap)["Parquet_go_root\x01Timestamp"].Values[2].(int64))
+	assert.Equal(t, int64(4001), (*tableMap)["Parquet_go_root\x01Timestamp"].Values[3].(int64))
+	assert.Equal(t, int64(5001), (*tableMap)["Parquet_go_root\x01Timestamp"].Values[4].(int64))
+	assert.Equal(t, int64(6001), (*tableMap)["Parquet_go_root\x01Timestamp"].Values[5].(int64))
+	assert.Equal(t, 6, len((*tableMap)["Parquet_go_root\x01Status"].Values))
+	assert.Equal(t, "RUNNING", (*tableMap)["Parquet_go_root\x01Status"].Values[0].(string))
+	assert.Equal(t, "ENQUEUED", (*tableMap)["Parquet_go_root\x01Status"].Values[1].(string))
+	assert.Equal(t, "COMPLETED", (*tableMap)["Parquet_go_root\x01Status"].Values[2].(string))
+	assert.Equal(t, "ERRORED", (*tableMap)["Parquet_go_root\x01Status"].Values[3].(string))
+	assert.Equal(t, "CANCELLED", (*tableMap)["Parquet_go_root\x01Status"].Values[4].(string))
+	assert.Equal(t, "UPSTREAM_NOT_PROCESSED", (*tableMap)["Parquet_go_root\x01Status"].Values[5].(string))
+
+	fmt.Println(tableMap)
+}
+
+func TestMarshalTestNestedElem(t *testing.T) {
+	testNestedElems := []interface{}{
+		testNestedElem{},
+		testNestedElem{SubElem: subElem{Val: "hi"}},
+		testNestedElem{SubPtr: &subElem{Val: "hi"}},
+		testNestedElem{SubList: []subElem{}},
+		testNestedElem{SubList: []subElem{{Val: "hi"}}},
+		testNestedElem{SubList: []subElem{{Val: "hi"}, {}, {Val: "there"}}},
+		testNestedElem{SubRepeated: []*subElem{}},
+		testNestedElem{SubRepeated: []*subElem{{Val: "hi"}}},
+		testNestedElem{SubRepeated: []*subElem{{Val: "hi"}, nil, {Val: "there"}}},
+		testNestedElem{EmptyPtr: &struct{}{}},
+		testNestedElem{EmptyList: []struct{}{}},
+		testNestedElem{EmptyList: []struct{}{{}}},
+		testNestedElem{EmptyList: []struct{}{{}, {}}},
+		testNestedElem{EmptyRepeated: []*struct{}{}},
+		testNestedElem{EmptyRepeated: []*struct{}{{}}},
+		testNestedElem{EmptyRepeated: []*struct{}{{}, nil, {}}},
+	}
+	schemaHandler, err := schema.NewSchemaHandlerFromProtoStruct(testNestedElem{})
+	if err != nil {
+		t.Errorf("failed to get schema handler: %v", err)
+	}
+	tableMap, err := MarshalProto(testNestedElems, schemaHandler)
+	if err != nil {
+		t.Errorf("failed to marshal values: %v", err)
+	}
+	assert.Equal(t, 8, len(schemaHandler.ValueColumns))
+	for i := 0; i < 18; i++ {
+		assert.Equal(t, nil, (*tableMap)["Parquet_go_root\x01EmptyRepeated\x01List\x01Element"].Values[i])
+	}
+	// There are total 16 objects in the list
+	assert.Equal(t, 16, len((*tableMap)["Parquet_go_root\x01SubElem\x01Val"].Values))
+	assert.Equal(t, "hi", (*tableMap)["Parquet_go_root\x01SubElem\x01Val"].Values[1].(string))
```

Comment:
NIT: Please extract into a constant/intermediate variable to avoid duplicating literals.